Introducing the jira software plugin for jenkins (#2454)
jenkins-infra/jenkins.io: content/blog/2019/08/2019-08-30-introducing-the-jira-software-plugin-for-jenkins.adoc

---
layout: post
title: "Introducing the Jira Software plugin for Jenkins"
tags:
- jira
- plugin
- pipeline
author: rafalmyslek
---

> NOTE: This is a guest post by Rafal Myslek from link:https://www.atlassian.com/jira[Atlassian].

Reviewer: Not sure this note is needed in this case

According to a link:https://www.atlassian.com/blog/software-teams/modern-software-development-trends[recent survey] we conducted, software & IT teams on average use 4+ tools to move code from development to customer-facing production. As a result, teams struggle with keeping the status of work updated and understanding the overall health of their delivery pipeline.

> To solve this problem, I am excited to announce that we built link:https://plugins.jenkins.io/atlassian-jira-software-cloud[an official plugin between Jenkins and Jira Software Cloud]. The plugin automatically associates build and deployment information from Jenkins with relevant Jira issues and exposes key information about your pipeline across Jira issues, boards and via JQL. This means you can use Jira Software to automatically update and track issues through your complete development pipeline, from backlog to release.

Reviewer: ```suggestion
To solve this problem, I am excited to announce that we built link:https://plugins.jenkins.io/atlassian-jira-software-cloud[an official Jenkins plugin for Jira Software Cloud]. The plugin automatically associates build and deployment information from Jenkins with relevant Jira issues and exposes key information about your pipeline across Jira issues, boards and via JQL. This means you can use Jira Software to automatically update and track issues through your complete development pipeline, from backlog to release.
```

[source]
----
deploymentEnvironmentType ~ “production“ AND development[pullrequests].open
----

== Get started

*In Jira Software Cloud*

*Create OAuth credentials in Jira for Jenkins*

. Navigate to *Jira home > Jira settings > Apps*.
. Select *OAuth credentials*.
. Select *Create credentials*.
. Enter the following details:
* _App name_ - Jenkins
* _App logo_ - A URL to the Jenkins logo, which will be used as an icon in the list of credentials. Eg: https://jenkins.yourcompany.com/logo.png
* Server base URL - The URL to your Jenkins server. Eg: https://jenkins.yourcompany.com

*In Jenkins*

*Install the Jenkins plugin*

. Login to your Jenkins server and navigate to the Plugin Manager.

. Select the 'Available' tab and search for 'Atlassian Jira Software Cloud' as the plugin name then install it.

> * The open-source plugin is hosted at on the Jenkins GitHub account. link:https://github.com/jenkinsci/atlassian-jira-software-cloud-plugin[You can check it out here].

Reviewer 2: Extra words possibly?
```suggestion
* The open-source plugin is hosted in the Jenkins GitHub account. link:https://github.com/jenkinsci/atlassian-jira-software-cloud-plugin[You can check it out here].
```

. From the left-hand menu, select *Pipeline Syntax*.
. In the Snippet Generator, select *jiraSendDeploymentInfo* or *jiraSendBuildInfo* from the dropdown list of Sample Steps and fill in the relevant details.
. Select *Generate Pipeline Script* and copy/paste the output into your _Jenkinsfile_ on the relevant Repository you are using. This will be used to notify Jira when you run that pipeline on that repo.

*For sending build information*

This is an example snippet of a very simple ‘build’ stage set up in a _Jenkinsfile_. After the pipeline is run, it will post the build information to your Jira Cloud site by looking at the branch name. If there is a Jira issue key (e.g. “TEST-123”) in the branch name, it will send the data over to Jira.

*Jenkinsfile example*

[source, groovy]
----
pipeline {
     agent any
     stages {
         stage('Build') {
             steps {
                 echo 'Building...'
             }
             post {
                 always {
                     jiraSendBuildInfo site: 'example.atlassian.net'
                 }
             }
         }
     }
 }
----

*For sending deployment information*

> This is an example snippet of two sages that runs on any change to the staging or master branch. Again, we use a post step to send deployment data to Jira and the relevant issues. Here, the *environmentId*, *environmentName*, and *environmentType* need to be set to whatever you want to appear in Jira.

Reviewer 2: Minor spelling fix.
```suggestion
This is an example snippet of two stages that run on any change to the staging or master branch. Again, we use a post step to send deployment data to Jira and the relevant issues. Here, the *environmentId*, *environmentName*, and *environmentType* need to be set to whatever you want to appear in Jira.
```

             steps {
                 echo 'Building...'
             }
             post {
                 always {
                     jiraSendBuildInfo site: 'example.atlassian.net'
                 }
             }
         }
         stage('Deploy - Staging') {
             when {
                 branch 'master'
             }
             steps {
                 echo 'Deploying to Staging from master...'
             }
             post {
                 always {
                     jiraSendDeploymentInfo site: 'example.atlassian.net', environmentId: 'us-stg-1', environmentName: 'us-stg-1', environmentType: 'staging'
                 }
             }
         }
         stage('Deploy - Production') {
            when {
                branch 'master'
            }
            steps {
                echo 'Deploying to Production from master...'
            }
            post {
                always {
                    jiraSendDeploymentInfo site: 'example.atlassian.net', environmentId: 'us-prod-1', environmentName: 'us-prod-1', environmentType: 'production'
                }
            }
         }
     }
 }
----

== Questions or feedback?

> If you have any questions, please link:https://support.atlassian.com/contact/[Atlassian support] and they will route it to the correct team to help you.

Reviewer 2: Missing a word in the sentence?
```suggestion
If you have any questions, please contact link:https://support.atlassian.com/contact/[Atlassian support] and they will route it to the correct team to help you.
```
